components/script/dom/htmlformelement.rs

```diff
@@ -1332,6 +1390,29 @@ impl<'a> FormSubmitter<'a> {
                 ),
         }
     }
+
+    // https://html.spec.whatwg.org/multipage/#concept-submit-button
+    fn is_submit_button(&self) -> bool {
+        match *self {
+            FormSubmitter::InputElement(input_element) => {
+                input_element.input_type() == InputType::Submit ||
+                    input_element.input_type() == InputType::Image
+            },
+            FormSubmitter::ButtonElement(button_element) => {
```

Comment:
and I think this would be https://html.spec.whatwg.org/multipage/#submit-button-state-(type=submit)

Reply:
I believe this is https://html.spec.whatwg.org/multipage/#attr-button-type-submit-state, added it